api/proto/inference_service.proto

```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
+
+package api;
+option go_package = "github.com/onepanelio/core/api/gen";
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+
+service InferenceService {
+  rpc CreateInferenceService (CreateInferenceServiceRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/apis/v1beta1/{namespace}/inferenceservice"
+      body: "*"
+    };
+  }
+
+  rpc GetInferenceService(InferenceServiceIdentifier) returns (GetInferenceServiceResponse) {
+    option (google.api.http) = {
+      get: "/apis/v1beta1/{namespace}/inferenceservice/{name}"
+    };
+  }
+
+  rpc DeleteInferenceService (InferenceServiceIdentifier) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/apis/v1beta1/{namespace}/inferenceservice/{name}"
+    };
+  }
+}
+
+message InferenceServiceIdentifier {
+  string namespace = 1;
+  string name = 2;
+}
+
+message Env {
+    string name = 1;
+    string value = 2;
+}
+
+message Container {
+    string image = 1;
+    string name = 2;
+    repeated Env env = 3;
+}
+
+message Transformer {
+  repeated Container containers = 1;
+  string minCpu = 2;
+  string minMemory = 3;
+  string maxCpu = 4;
+  string maxMemory = 5;
+}
+
+message InferenceServicePredictor {
+  string name = 1;
+  string runtimeVersion = 2;
+  string storageUri = 3;
+  string nodeSelector = 4;
+  string minCpu = 5;
+  string minMemory = 6;
+  string maxCpu = 7;
+  string maxMemory = 8;
+}
+
+message CreateInferenceServiceRequest {
+  string namespace = 1;
+  string name = 2;
+  string transformerImage = 3;
```

Comment:
Is there a reason we are exposing only `transformerImage`? Seems a bit inconsistent.

Reply:
@rushtehrani That's for convenience. Otherwise you have to specify a transformer, with containers, with envs.

Reply:
@Vafilor got it, edited my original comment before seeing your reply.

Reply:
@rushtehrani Sure, that works.